pkg/utils/k8sutil.go

```
@@ -847,3 +658,278 @@ func DeleteIngress(client kubernetes.Interface, name, ns string) error {
+				},
+			},
+		},
+	})
+	if len(dpm.Spec.Template.Spec.Containers) == 0 {
+		dpm.Spec.Template.Spec.Containers = append(dpm.Spec.Template.Spec.Containers, v1.Container{})
+	}
+
+	modName, handlerName, err := splitHandler(funcObj.Spec.Handler)
+	if err != nil {
+		return err
+	}
+
+	imageName, _, _, err := GetFunctionData(funcObj.Spec.Runtime, funcObj.Spec.Type, modName)
+	if err != nil {
+		return err
+	}
+
+	dpm.Spec.Template.Spec.Containers[0].Image = imageName
+	dpm.Spec.Template.Spec.Containers[0].Name = funcObj.Metadata.Name
+	dpm.Spec.Template.Spec.Containers[0].Ports = append(dpm.Spec.Template.Spec.Containers[0].Ports, v1.ContainerPort{
+		ContainerPort: 8080,
+	})
+	dpm.Spec.Template.Spec.Containers[0].Env = append(dpm.Spec.Template.Spec.Containers[0].Env,
+		v1.EnvVar{
+			Name:  "FUNC_HANDLER",
+			Value: handlerName,
+		},
+		v1.EnvVar{
+			Name:  "MOD_NAME",
+			Value: modName,
+		},
+		v1.EnvVar{
+			Name:  "TOPIC_NAME",
+			Value: funcObj.Spec.Topic,
+		})
+	dpm.Spec.Template.Spec.Containers[0].VolumeMounts = append(dpm.Spec.Template.Spec.Containers[0].VolumeMounts, v1.VolumeMount{
+		Name:      funcObj.Metadata.Name,
+		MountPath: "/kubeless",
+	})
+
+	// update deployment for custom runtime
+	if funcObj.Spec.Deps != "" {
+		updateDeployment(dpm, funcObj.Spec.Runtime)
+		//TODO: remove this when init containers becomes a stable feature
+		addInitContainerAnnotation(dpm)
+	}
+
+	// add liveness Probe to deployment
+	if funcObj.Spec.Type != pubsubFunc {
+		livenessProbe := &v1.Probe{
+			InitialDelaySeconds: int32(3),
+			PeriodSeconds:       int32(30),
+			Handler: v1.Handler{
+				HTTPGet: &v1.HTTPGetAction{
+					Path: "/healthz",
+					Port: intstr.FromInt(8080),
+				},
+			},
+		}
+		dpm.Spec.Template.Spec.Containers[0].LivenessProbe = livenessProbe
+	}
+
+	_, err = client.Extensions().Deployments(funcObj.Metadata.Namespace).Create(dpm)
+	if err != nil && k8sErrors.IsAlreadyExists(err) {
+		data, _ := json.Marshal(dpm)
```

we shouldn't ignore errors here